libasn1compiler/asn1c_C.c

```diff
@@ -2421,9 +2436,10 @@ try_inline_default(arg_t *arg, asn1p_expr_t *expr, int out) {
 		//expr->marker.flags &= ~EM_INDIRECT;
 		return 0;
 	default:
-	  if(etype & ASN_STRING_KM_MASK) {
-		if(expr->marker.default_value == NULL
-		|| expr->marker.default_value->type != ATV_STRING)
+	  if(expr->marker.default_value) {
+	    if(!(etype & ASN_STRING_KM_MASK) &&
```

Comment:
Are you sure it should be `&&` and not `||`? The logic seems to need "if the type is not a STRING derivative, *or* default not provided, *or* (default is neither `BIT STRING` nor `OCTET STRING`), then break". Correct?

Reply:
```
BIT STRING -> etype : ASN_BASIC_BIT_STRING , default_value_type : ATV_BITVECTOR
OCTET STRING -> etype : ASN_BASIC_OCTET_STRING, default_value_type : ATV_BITVECTOR
IA5String -> etype : ASN_STRING_IA5String, default_value_type : ATV_STRING
```
However, ASN_BASIC_BIT_STRING and ASN_BASIC_OCTET_STRING do not have bits set with ASN_STRING_KM_MASK.

I admit that this is a little obscure but I tried to be safe for possible combinations I could figure out.
To simplify it, perhaps we could just check `default_value->type` against ATV_STRING and ATV_BITVECTOR.

Reply:
I see. Would you say that `ATV_STRING` and `ATV_BITSTRING` cover all the possibilities we want to include here?